### ISSUE TITLE
In 2.10.x AnyUrl does not validate data uris

### ISSUE THREAD

**Issue by dolfim-ibm** [pydantic/pydantic#10968]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

In pydantic 2.10.0 and 2.10.1 the `AnyUrl` does not accept data uris anymore. The example code works with Pydantic 2.9.2 but fails with the newer versions with the following error:

```
pydantic_core._pydantic_core.ValidationError: 1 validation error for function-wrap[wrap_val()]
  Input should be a valid URL, empty host [type=url_parsing, input_value='data:image/png;base64,iV...Y4OHwAAAABJRU5ErkJggg==', input_type=str]

```

### Example Code

```Python
from pydantic import AnyUrl

data_uri = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg=="
url = AnyUrl(data_uri)
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.10.1
        pydantic-core version: 2.27.1
          pydantic-core build: profile=release pgo=false
                 install path: /private/tmp/pyd/venv/lib/python3.12/site-packages/pydantic
               python version: 3.12.7 (main, Oct  1 2024, 02:05:46) [Clang 15.0.0 (clang-1500.3.9.4)]
                     platform: macOS-15.1-arm64-arm-64bit
             related packages: pydantic-settings-2.6.1 typing_extensions-4.12.2
                       commit: unknown
```
```

**Comment by sydney-runkle** [comment]
Hi!

I believe this is better behavior - `AnyUrl` documentation suggested (even in v2.9.2) that a `host` was required, and now that is enforced.

Let me chat with the team and see if this is something we should have changed in the other direction...

**Comment by sydney-runkle** [comment]
Alright, we'll fix this for v2.10.2. Thanks for bringing this to our attention!
